File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String temp=s.toLowerCase();
        String temp1="";
        for(int i=0;i<s.length();i++){
            if(temp.charAt(i)>=97&&temp.charAt(i)<=122||temp.charAt(i)>=48&&temp.charAt(i)<=57){
                temp1=temp1+temp.charAt(i);

            } else{
                continue;
            }
        }
        String second=temp1;
        System.out.println(temp1);
        System.out.println(second);
        for(int i=0;i<second.length();i++){
            if(second.charAt((second.length()-1)-i)!=temp1.charAt(i)){
                return false;
            }
        }
        
        return true;


    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        string n;
        for(int i=0;i<s.length();i++){
            char c=s[i];
            if (isalnum(c)) {
                n += tolower(c);
            }
        }
        string rev = n;
        
        reverse(rev.begin(), rev.end());
         
        return rev == n;
    }
};